compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/TailrecLowering.kt

```diff
@@ -130,31 +127,11 @@ private class BodyTransformer(
             +irSetVar(parameterToVariable[parameter]!!.symbol, argument)
         }
 
+        // Unspecified arguments do not exist; Calls to, e.g. foo, are replaced with foo$default.
```

Comment:
This piece of code is totally unclear: foo$default always calls foo function
With tailrec optimization foo$default call from foo should be avoided